packages/plugins/src/SidebarPlugin.ts

```diff
@@ -218,18 +223,20 @@ const SidebarPlugin: PluginType<SidebarPluginPage, SidebarPluginOptions, Sidebar
       const removeExcludedPages = page => !(page.sidebar && page.sidebar.exclude);
 
       function sortPagesByPriority(sidebarData) {
-        const pagesByPriority = sidebarData.map(page => {
-          if (page.childNodes.length > 1) {
-            const pagesByPriority = page.childNodes.sort(
-              (pageA, pageB) =>
-                (pageB.priority ? pageB.priority : -1) - (pageA.priority ? pageA.priority : -1)
-            );
-            sortPagesByPriority(page.childNodes);
-            return { ...page, childNodes: pagesByPriority };
-          } else {
-            return page;
-          }
-        });
+        const pagesByPriority = sidebarData
+          .filter(page => !!page) // filter out null or undefined values
```

Comment:
this feels like we might be filtering out null or undefined values but should they be there, to begin with?
We are potentially masking an error and not fixing it?

here is a thought.

If the pageExtensions is not set then would it glob every file in the directory, rather than just .mdx.
We have set the default value for pageExtensions to [], so it does not error but would glob every file and not just mdx.

Then if you have a file that does not contain frontmatter (or fullPath) it would be undefined.

So we could check what the file glob returns from this file and ensure every file has a fullPath and is a mdx file.

I know JP was keen to not hardcode mdx as a default type as they could be markdown.
Potentially something like  is needed in our standard generator configs`.{md,mdx}`

Reply:
This is all theoritical ... but if true,  explains why sometimes it works and sometimes it does not, because it depends what files are in the directory being globbed.

Reply:
> This is all theoritical ... but if true, explains why sometimes it works and sometimes it does not, because it depends what files are in the directory being globbed.

Extensions is set in the source, do you think it is not being applied?